network/p2p/middleware.go

```diff
@@ -179,7 +167,9 @@ func NewMiddleware(
 			ready()
 
 			<-ctx.Done()
-			mw.stop()
+
+			// wait for the readConnection and readSubscription routines to stop
+			mw.wg.Wait()
```

Comment:
Could we add some log how long this wait will take?

I'm a bit concerned this Wait will take too long.

Reply:
sure. this was taken directly from the previous version though, so there's no change in behavior

Reply:
Can you point out where is the code from the pervious version?

Github shows the change as changing from `mv.stop()` to `mv.wg.Wait()`.  So I thought the logic here is changed, the mv is now stopped elsewhere, and here, we are simply waiting for the result of the `stop`

Reply:
the original call was here:
https://github.com/onflow/flow-go/pull/3142/files#diff-95120943dbf1b8334da6c35dc229d380cea1f66bb6751a652aab627b9082ffa8L352

This PR converts the peer manager and libp2p node to components, so middleware no longer needs to manage their shutdown. `stop()` became just that single line so I removed the method